### PR TITLE
Improvement: Prem Templates

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR adds an improvement to `dsp.modules.prem.PremAI` by introducing using Prem Prompt Templates throughout dspy calls.

### CODE REVIEW

File: dsp/modules/premai.py [hunk 1]
```diff
@@ -122,8 +120,31 @@ def _get_all_kwargs(self, **kwargs) -> Dict[str, Any]:
...
+                raise TypeError("params must be a dictionary")
+
+            messages = [
+                {
+                    "role": "user",
+                    "template_id": template_id,
+                    "params": params,
+                },
+            ]
+            all_kwargs.pop("template_id")
```
Comment (arnavsinghvi11) [inline dsp/modules/premai.py:146]: any reason to remove the params from the kwargs which are propagated in history? might be relevant to see how the DSPy prompt was changed in addition to the Prem template but I'll defer to you here.

**Comment by arnavsinghvi11** [comment]
Thanks @Anindyadeep ! left a minor comment and revised some of the documentation - should be good to merge after that!

**Comment by Anindyadeep** [comment]
> Thanks @Anindyadeep ! left a minor comment and revised some of the documentation - should be good to merge after that!

Yes you are right, so I added some latest changes addressing those. Additionally I combined the raw and all kwargs, just to remove redundancy.

**Comment by arnavsinghvi11** [comment]
Thanks @Anindyadeep !

**Pull request merged**